synapse/handlers/sync.py

```diff
@@ -747,7 +747,7 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
                     if types:
```

Comment:
I'm struggling to remember what this block is for (it's a shame there's no comment). AIUI, this is dealing with the situation that:

* We're returning an incremental sync, with no "gap" since the previous sync, so normally there would be no state to return
* But we're lazy-loading, so the client might need some more member events to understand the events in this timeline
* So we fish out all the member events corresponding to the timeline here, and then dedupe them below

Could you confirm and add a comment?

Reply:
confirmed; have added comment.